crates/interface/src/types.rs

```diff
@@ -133,9 +133,7 @@ pub struct BlockContext {
     /// Source of randomness, currently holds the value
     /// of prevRandao.
     pub mix_hash: U256,
-    /// Version of the protocol.
-    /// It's used to determine the implementation of ZKsync OS and its config.
-    pub protocol_version: u32,
+    pub zksync_os_execution_version: u32,
```

Comment:
Maybe just `execution_version`? `zksync_os` prefix is redundant IMHO as that's implied by being in this crate/repo. I would add/keep the comment instead that makes it clear that this refers to the version of ZKsync OS.

Thinking more though - why do we even have this value in this crate? Having a "version" field in the interface meant to be the same for all versions sounds weird to me. If this was done purely for (de)serialization within server IMHO this can be trivially solved on the server side

Reply:
> Having a "version" field in the interface meant to be the same for all versions sounds weird to me

Meant to be different for version. 
Or course we can solve this on the server side by adding struct like BlockContextWithVersion and pass it around. Right now code is easy, we only use interface::BlockContext and before it's passed to zksync_os crates we convert it to BlockMetadataFromOracle. If we add BlockContextWithVersion it will be more complicated and IMO it's not justified at all